plugins/vpc-branch-eni/plugin/commands.go

```diff
@@ -280,9 +280,18 @@ func (plugin *Plugin) createVLANLink(branch *eni.Branch, linkName string, ipAddr
 			return err
 		}
 
+		// If gatewayIPAddress is nil, that means the gateway ip is not provided in NetConfig. In that case,
+		// use the gateway address inferred from the subnet.
+		if gatewayIPAddress == nil {
```

Comment:
IIUC, you don't even need the `subnet, err := vpc.NewSubnet(vpc.GetSubnetPrefix(ipAddress))` block when `gatewayIPAddress` is provide (not `nil`). If that's correct, please refactor this to look something like this:

```
if gatewayIPAddress == nil {
 // Parse VPC subnet.
 subnet, err := vpc.NewSubnet(vpc.GetSubnetPrefix(ipAddress))
 ..
 gatewayIPAddress = subnet.Gateways[0]
 log.Infof("..")
}

// Add default route via branch link.
route := &netlink.Route{
 ..
}
...
```

Reply:
updated